### PR TITLE
Add search by metadata to orders and checkouts

### PR DESCRIPTION
**Pull request by iprlic**
I want to merge this change because...

Add search by metadata to orders and checkouts

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/order/filters.py [hunk 1]
```diff
@@ -77,15 +77,15 @@ def filter_order_search(qs, _, value):
         return filter_order_by_payment(qs, payment_id)
 
     users = User.objects.filter(
-        Q(email__trigram_similar=value)
-        | Q(first_name__trigram_similar=value)
-        | Q(last_name__trigram_similar=value)
+        Q(email__ilike=value) | Q(first_name__ilike=value) | Q(last_name__ilike=value)
```
Comment (iprlic) [inline saleor/graphql/order/filters.py:80]: Changed as it was suggested in my previous URL as well

File: saleor/graphql/checkout/filters.py [hunk 2]
```diff
@@ -68,6 +68,8 @@ def filter_checkout_search(qs, _, value):
 
     filter_option = Q(Exists(users.filter(id=OuterRef("user_id"))))
 
+    filter_option |= Q(metadata__icontains=value)
```
Comment (tomaszszymanski129) [inline saleor/graphql/checkout/filters.py:71]: `__icontains` is a performance killer, use `__ilike` instead.

File: saleor/graphql/order/filters.py [hunk 3]
```diff
@@ -77,15 +77,15 @@ def filter_order_search(qs, _, value):
...
     ).values("pk")
 
-    filter_option = Q(user_email__trigram_similar=value) | Q(
+    filter_option = Q(user_email__ilike=value) | Q(
         Exists(users.filter(pk=OuterRef("user_id")))
     )
 
+    filter_option |= Q(metadata__icontains=value)
```
Comment (tomaszszymanski129) [inline saleor/graphql/order/filters.py:87]: same here, change to `__ilike`.

File: saleor/graphql/checkout/filters.py (hunk 2 above)
Comment (iprlic) [reply saleor/graphql/checkout/filters.py:71]: This is jsonb field, can't do ilike on it, AFAIK?!

**Comment by maarcingebala** [comment]
Closing as we agreed that dedicated `metadata` filter should be used and these changes may cause performance issues.

**Pull request closed without merging**